File: src/main/java/net/tianben/advancedcombat/enchated/EnchantedSwordItem.java
package net.tianben.advancedcombat.enchated;

import net.minecraft.item.ItemStack;
import net.minecraft.item.SwordItem;
import net.minecraft.item.ToolMaterial;

public class EnchantedSwordItem extends SwordItem {
    public EnchantedSwordItem(ToolMaterial toolMaterial, int attackDamage, float attackSpeed, Settings settings) {
        super(toolMaterial, attackDamage, attackSpeed, settings);
    }

    @Override
    public boolean hasGlint(ItemStack stack) {
        return true;
    }
}

File: src/main/kotlin/net/tianben/advancedcombat/enchanted/EnchantedSwordItem.kt
package net.tianben.advancedcombat.enchanted

import net.minecraft.item.ItemStack
import net.minecraft.item.SwordItem
import net.minecraft.item.ToolMaterial

class EnchantedSwordItem(toolMaterial: ToolMaterial, attackDamage: Int, attackSpeed: Float, settings: Settings) :
    SwordItem(toolMaterial, attackDamage, attackSpeed, settings) {
    override fun hasGlint(stack: ItemStack?): Boolean {
        return true
    }
}

